coalib/coala.py

```diff
@@ -59,11 +60,35 @@ def main(debug=False):
 
         if args.show_bears:
             from coalib.settings.ConfigurationGathering import (
-                get_filtered_bears)
-
-            local_bears, global_bears = get_filtered_bears(
-                args.filter_by_language, log_printer)
-
+                get_all_bears)
+            filtered_bears = get_all_bears(log_printer)
+            if args.filter_by_language:
+                # filter_by_language option should be deprecated soon.
+                # Since it is already covered by filter_by option.
+                log_printer.warn(
+                    "'--filter-by-language ...' option is deprecated. "
+                    "Use '--filter-by language ...' option instead")
+                args.filter_by = args.filter_by or []
+                args.filter_by.append(['language'] + args.filter_by_language)
+            if args.filter_by:
+                # Each iteration of the following loop applies
+                # filters one by one provided as arguments
+                for filter_by in args.filter_by:
+                    filter_by_x, *filter_args = filter_by
+                    if len(filter_args) == 0:
+                        continue
+                    try:
+                        filtered_bears = FilterHelper.get_filtered_bears(
+                            filter_by_x, filter_args, filtered_bears)
```

Comment:
Here, the invocation order of filters won't give different results.
If I understood your comment correctly,
In your opinion:
`coala -B --filter-by language c --filter-by can_fix syntax`
and
`coala -B --filter-by can_fix syntax --filter-by language c`
can give different results.

But, as per the implementation, the result is nothing but a set of bears containing intersection of all the bears filtered out from each filters individually. However, implementation is not directly taking intersection, but it doing something like folding. But the essence is same. We just want to show the intersection of all the filters.

Correct me if I'm wrong.